### ISSUE TITLE
Inversed precision & recall

### ISSUE THREAD

**Issue by YaYaB** [jolibrain/deepdetect#746]
If Ok, please give as many details as possible to help us solve the problem more efficiently.
#### Configuration
- Version of DeepDetect:
    - [X] Locally compiled on:
        - [X] Ubuntu 14.04 LTS
        - [ ] Mac OSX
        - [ ] Other:
    - [ ] Docker
    - [ ] Amazon AMI
- Commit (shown by the server when starting):
88e93254ead67a8032166e98af3e46837fbba039

#### Your question / the problem you're facing:
I was trying to train a specific model. However after finishing the training and launching some predictions on the test set I saw that the metrics during the trainings were a bit weird. I used sklearn to compute precision and recall and saw that the metrics precision and recall are inverted.
I have checked the code and indeed if you check [here](https://github.com/jolibrain/deepdetect/blob/badb7d03d760fe229526391da7c2a5a740890a65/src/supervisedoutputconnector.h#L1609-L1610) the names do not correspond.
```
 dMat conf_csum = conf_matrix.colwise().sum()
```
will compute the sum for reach row, and
```
 dMat conf_rsum = conf_matrix.rowwise().sum();
```
will compute the sum for reach row.
However those are used to compute precision and recall in [here](https://github.com/jolibrain/deepdetect/blob/badb7d03d760fe229526391da7c2a5a740890a65/src/supervisedoutputconnector.h#L1613-L1614)

I can prepare an MR that would fix this easily:
```
dMat conf_rsum = conf_matrix.colwise().sum()
dMat conf_csum = conf_matrix.rowwise().sum();
```

Moreoever I saw that I was not able to replicate the f1-score metric computed in DD.
In [here](https://github.com/jolibrain/deepdetect/blob/badb7d03d760fe229526391da7c2a5a740890a65/src/supervisedoutputconnector.h#L1615-L1618).
We see that the f1-score is computed in the global precision / recall of the model. However I think it would be better to compute the mean f1-score for each class to compute the global f1-score (as sklearn does with the average "macro" or "weighted").
In the same time getting more metrics (precision, recall, f1score for each class) would be nice.

It is a bit complicated to replicate this with a model as I would need to train a model and give you access to data to test it.
You can try all of this with mnist or something like this just to see that the metrics seem wrong.

**Comment by fantes** [comment]
Hi

thank you for the bug report. This PR should fix things up: 
https://github.com/jolibrain/deepdetect/pull/757

moreover, with this PR ,  if you add "f1full" to your `measure` list (or replace "f1" by "f1full") dede will output per class precision recall and f1.

**Comment by fantes** [comment]
let me know if this seems ok for your needs

**Comment by YaYaB** [comment]
Sounds perfect for me!
